src/cli.py

```diff
@@ -0,0 +1,111 @@
+from datetime import date, timedelta, datetime
+from typing import List, Optional
+
+from src.browser import create_browser_driver, ACCEPTED_BROWSERS
+from src.constants import SUPPORTED_OUTPUT_EXTENSIONS, TEXT_SEARCH_FILING_CATEGORIES_MAPPING
+from src.text_search import EdgarTextSearcher
+
+
+def _validate_args(
+        search_keywords: List[str],
+        start_date: date,
+        end_date: date,
+        filing_type: Optional[str],
+        min_wait_secs: float,
+        max_wait_secs: float,
+        stop_after_n: int,
+        browser_name: str,
+        destination: str,
+) -> None:
+    """
+    Validate the CLI arguments, raises an error if the arguments are invalid.
+    """
+
+    if not search_keywords:
+        raise ValueError("At least one search keyword is required")
+    if start_date > end_date:
+        raise ValueError("start_date cannot be after end_date")
+    if min_wait_secs <= 0:
+        raise ValueError("wait_for_request_secs cannot be negative or null")
+    if max_wait_secs < min_wait_secs:
+        raise ValueError("max_wait_secs cannot be less than min_wait_secs")
+    if stop_after_n < 0:
+        raise ValueError("stop_after_n cannot be negative")
```

Comment:
If a user provides `retries`, the name change in the error might be confusing, suggest refactoring `stop_after_n` to `retries`